indico/modules/events/contributions/operations.py

```diff
@@ -153,3 +155,28 @@ def delete_subcontribution(subcontrib):
     logger.info('Subcontribution %s deleted by %s', subcontrib, session.user)
     subcontrib.event_new.log(EventLogRealm.management, EventLogKind.negative, 'Subcontributions',
                              'Subcontribution "{}" has been deleted'.format(subcontrib.title), session.user)
+
+
+@no_autoflush
+def create_contribution_from_abstract(abstract, contrib_session=None):
+    event = abstract.event_new
+    contrib_person_links = set()
+    person_link_attrs = {'_title', 'address', 'affiliation', 'first_name', 'last_name', 'phone', 'author_type',
+                         'is_speaker'}
+    for abstract_person_link in abstract.person_links:
+        link = ContributionPersonLink(person=abstract_person_link.person)
+        link.populate_from_attrs(abstract_person_link, person_link_attrs)
+        contrib_person_links.add(link)
+
+    duration = contrib_session.default_contribution_duration if contrib_session else timedelta(minutes=15)
```

Comment:
I think having a single default duration somewhere (or a setting.. maybe even be a contribution type option, since that's what often determines the duration of the contribution) would be much cleaner, but doesn't need to be done in this PR. It's more a nice-to-have thing.